cram/cram_io.c

```diff
@@ -1811,41 +1810,58 @@ static refs_t *refs_load_fai(refs_t *r_orig, const char *fn, int is_err) {
         if (!(r = refs_create()))
             goto err;
 
-    /* Open reference, for later use */
-    if (stat(fn, &sb) != 0) {
-        if (is_err)
-            perror(fn);
-        goto err;
-    }
-
     if (r->fp)
         if (bgzf_close(r->fp) != 0)
             goto err;
     r->fp = NULL;
 
-    if (!(r->fn = string_dup(r->pool, fn)))
-        goto err;
-
-    if (fn_l > 4 && strcmp(&fn[fn_l-4], ".fai") == 0)
-        r->fn[fn_l-4] = 0;
+    /* Look for a FASTA##idx##FAI format */
+    char *fn_delim = strstr(fn, HTS_IDX_DELIM);
+    if (fn_delim) {
+        char *fn_tmp = strdup(fn);
```

Comment:
Instead of making a copy here, I think `string_ndup(r->pool, fn, fn_delim - fn)` should work.